File: internal/tui/editor.go
// editor.go builds the exec.Cmd used to open a ticket file in an external
// editor. The editor is resolved in order: user config → $EDITOR env var → vi.
package tui

import (
	"os"
	"os/exec"
	"strings"
)

func editorCommand(path, preferred string) *exec.Cmd {
	editor := preferred
	if editor == "" {
		editor = os.Getenv("EDITOR")
	}
	if editor == "" {
		editor = "vi"
	}
	parts := strings.Fields(editor)
	bin, err := exec.LookPath(parts[0])
	if err != nil {
		bin = parts[0]
	}
	args := append(parts[1:], path)
	return exec.Command(bin, args...)
}

File: src/tui/editor.rs
use std::path::{Path, PathBuf};

/// Return the editor binary + args to open `path`.
/// Resolution order: $EDITOR env → "vi".
/// Args are parsed as shell words so quoted args work (TUI-08).
pub fn editor_path(ticket: &Path) -> (PathBuf, Vec<String>) {
    let editor = std::env::var("EDITOR").unwrap_or_else(|_| "vi".to_owned());
    let mut parts = shell_words::split(&editor).unwrap_or_else(|_| vec!["vi".to_owned()]);
    if parts.is_empty() {
        parts.push("vi".to_owned());
    }
    let bin = PathBuf::from(parts.remove(0));
    let mut args = parts;
    args.push(ticket.to_string_lossy().into_owned());
    (bin, args)
}

/// Launch the editor synchronously; terminal must already be restored.
pub fn open(ticket: &Path) {
    let (bin, args) = editor_path(ticket);
    let _ = std::process::Command::new(&bin).args(&args).status();
}
